storm-core/src/jvm/org/apache/storm/grouping/ShuffleGrouping.java

```diff
@@ -35,7 +36,7 @@
 
     @Override
     public void prepare(WorkerTopologyContext context, GlobalStreamId stream, List<Integer> targetTasks) {
-        random = new Random();
+        random = new XORShiftRandom();
```

Comment:
Same as above.  Having repeated numbers is not a big deal here, but we do want to have a comment declaring that this is safe and what the ramifications are.